src/worker.cc

```diff
@@ -616,37 +620,50 @@ void from_json(rapidjson::Document& doc, Options& options) {
 
   // date_time
   auto date_time_type = rapidjson::get_optional<unsigned int>(doc, "/date_time/type");
-  auto date_time_value = rapidjson::get_optional<std::string>(doc, "/date_time/value");
   if (date_time_type && Options::DateTimeType_IsValid(*date_time_type)) {
     // check the type is in bounds
     auto const v = static_cast<Options::DateTimeType>(*date_time_type);
     if (v >= Options::DateTimeType_ARRAYSIZE)
       throw valhalla_exception_t{163};
     options.set_date_time_type(static_cast<Options::DateTimeType>(v));
     // check the value exists for depart at and arrive by
+    auto date_time_value = v != Options::current
```

Comment:
this works the same but seemed simplier. if its a current time type the string is current other wise its the user provided string